mdal/mdal_date_time.hpp

```diff
@@ -0,0 +1,143 @@
+/*
+ MDAL - Mesh Data Abstraction Library (MIT License)
+ Copyright (C) 2019 Vincent Cloarec (vcloarec at gmail dot com)
+*/
+
+#ifndef MDAL_DATE_TIME_HPP
+#define MDAL_DATE_TIME_HPP
+
+// Macro for exporting symbols
+// for unit tests (on windows)
+#define MDAL_TEST_EXPORT MDAL_EXPORT
+
+#include <string>
+#include <vector>
+
+#include "mdal.h"
+
+namespace MDAL
+{
+
+  class MDAL_TEST_EXPORT Duration
+  {
+    public:
+      enum Unit
+      {
+        milliseconds = 0,
+        seconds,
+        minutes,
+        hours,
+        days,
+        weeks,
+        months_CF,
+        exact_years
+      };
+
+      Duration();
+
+      ///TODO :  move operation
+
+      Duration( double duration, Unit unit );
+      Duration( const Duration &other );
+
+      double value( Unit unit ) const;
+
+      Duration &operator=( const Duration &other );
+      Duration operator-( const Duration &other ) const;
+      Duration operator+( const Duration &other ) const;
+      Duration &operator+=( const Duration &other );
+      Duration &operator-=( const Duration &other );
+      bool operator==( const Duration &other ) const;
+      bool operator!=( const Duration &other ) const;
+      bool operator<( const Duration &other ) const;
+      bool operator>( const Duration &other ) const;
+      bool operator>=( const Duration &other ) const;
+      bool operator<=( const Duration &other ) const;
+
+    private:
+      Duration( int64_t ms );
+      int64_t mDuration; //in ms
+
+      friend class DateTime;
```

Comment:
Not sure... those classes are "real friends" and I don't want to expose `int64_t mDuration` used for operation between `DateTime` and `Duration`. `int64_t mDuration` is now used to store the information but maybe, one day, it will not be anymore (for example, if we used a third library). So I don't want to expose this value.